/home/st6322041/anaconda3/envs/ML/bin/python3
/home/st6322041/anaconda3/envs/ML/lib/python3.9/site-packages/huggingface_hub/file_download.py:943: FutureWarning: `resume_download` is deprecated and will be removed in version 1.0.0. Downloads always resume when possible. If you want to force a new download, use `force_download=True`.
  warnings.warn(
The `load_in_4bit` and `load_in_8bit` arguments are deprecated and will be removed in the future versions. Please, pass a `BitsAndBytesConfig` object in `quantization_config` argument instead.
Downloading shards:   0%|          | 0/2 [00:00<?, ?it/s]/home/st6322041/anaconda3/envs/ML/lib/python3.9/site-packages/huggingface_hub/file_download.py:943: FutureWarning: `resume_download` is deprecated and will be removed in version 1.0.0. Downloads always resume when possible. If you want to force a new download, use `force_download=True`.
  warnings.warn(
Downloading shards:  50%|█████     | 1/2 [01:52<01:52, 112.10s/it]Downloading shards: 100%|██████████| 2/2 [02:32<00:00, 69.97s/it] Downloading shards: 100%|██████████| 2/2 [02:32<00:00, 76.29s/it]
Traceback (most recent call last):
  File "/home/st6322041/anaconda3/envs/ML/lib/python3.9/site-packages/transformers/utils/import_utils.py", line 1510, in _get_module
    return importlib.import_module("." + module_name, self.__name__)
  File "/home/st6322041/anaconda3/envs/ML/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/home/st6322041/anaconda3/envs/ML/lib/python3.9/site-packages/transformers/integrations/bitsandbytes.py", line 12, in <module>
    import bitsandbytes as bnb
  File "/home/st6322041/anaconda3/envs/ML/lib/python3.9/site-packages/bitsandbytes/__init__.py", line 11, in <module>
    from . import _ops, research, utils
  File "/home/st6322041/anaconda3/envs/ML/lib/python3.9/site-packages/bitsandbytes/_ops.py", line 15, in <module>
    register_fake = torch.library.impl_abstract
AttributeError: module 'torch.library' has no attribute 'impl_abstract'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/st6322041/Git/git/ML/finetuning.py", line 16, in <module>
    model = AutoModelForCausalLM.from_pretrained(
  File "/home/st6322041/anaconda3/envs/ML/lib/python3.9/site-packages/transformers/models/auto/auto_factory.py", line 563, in from_pretrained
    return model_class.from_pretrained(
  File "/home/st6322041/anaconda3/envs/ML/lib/python3.9/site-packages/transformers/modeling_utils.py", line 3564, in from_pretrained
    hf_quantizer.preprocess_model(
  File "/home/st6322041/anaconda3/envs/ML/lib/python3.9/site-packages/transformers/quantizers/base.py", line 182, in preprocess_model
    return self._process_model_before_weight_loading(model, **kwargs)
  File "/home/st6322041/anaconda3/envs/ML/lib/python3.9/site-packages/transformers/quantizers/quantizer_bnb_4bit.py", line 262, in _process_model_before_weight_loading
    from ..integrations import get_keys_to_not_convert, replace_with_bnb_linear
  File "<frozen importlib._bootstrap>", line 1055, in _handle_fromlist
  File "/home/st6322041/anaconda3/envs/ML/lib/python3.9/site-packages/transformers/utils/import_utils.py", line 1500, in __getattr__
    module = self._get_module(self._class_to_module[name])
  File "/home/st6322041/anaconda3/envs/ML/lib/python3.9/site-packages/transformers/utils/import_utils.py", line 1512, in _get_module
    raise RuntimeError(
RuntimeError: Failed to import transformers.integrations.bitsandbytes because of the following error (look up to see its traceback):
module 'torch.library' has no attribute 'impl_abstract'

real	2m37.805s
user	0m11.423s
sys	0m22.973s
